packages/block-library/src/image/edit.js

```diff
@@ -141,7 +142,12 @@ class ImageEdit extends Component {
 
 	componentDidUpdate( prevProps ) {
 		const { id: prevID, url: prevURL = '' } = prevProps.attributes;
-		const { id, url = '' } = this.props.attributes;
+		const { id, url = '', fileWidth } = this.props.attributes;
+		const imageData = this.props.image;
+		const blockWidth = getBlockWidth();
+
+		// Store the current block width inside the editor.
+		this.props.setAttributes( { editWidth: blockWidth } );
```

Comment:
A worry here is that if the value wasn't already assigned, it would immediately make the post "dirty" (prompting on navigation, triggering delayed autosave).

Reply:
Ideally we shouldn't call `setAttributes` in `componentDidUpdate` because it will trigger a second (ideally unnecessary render):

`[Something Triggers Update] -> [render] -> [componentDidUpdate] -> [setAttributes] -> [render] -> [componentDidUpdate]`

Reply:
Furthermore, there's nothing which would assign the `editWidth` attribute when the component first mounts?

Reply:
Hm, is there a better place/way to do this? We need it done once per inserted/uploaded image.